main.go

```diff
@@ -6,8 +6,11 @@ import (
 	"log"
 	"os"
 	"strings"
+	"errors"
 )
 
+var ErrorNoSuchKey = errors.New("No such key")
```

Comment:
Maybe `const` instead `var`?

Reply:
you cannot use `const` keyword for values returned from a function